src/Compilers/CSharp/Portable/Binder/Binder_Attributes.cs

```diff
@@ -147,10 +147,13 @@ private BoundAttribute BindAttributeCore(AttributeSyntax node, NamedTypeSymbol a
 
             HashSet<DiagnosticInfo> useSiteDiagnostics = null;
 
+            // Avoid argument errors causing cascading diagnostics from overload resolution
+            var overloadResolutionDiagnostics = analyzedArguments.HasErrors ? new DiagnosticBag() : diagnostics;
+
             // Bind attributeType's constructor based on the bound constructor arguments
             MethodSymbol attributeConstructor = attributeTypeForBinding.IsErrorType() ?
                 null :
-                BindAttributeConstructor(node, attributeTypeForBinding, analyzedArguments.ConstructorArguments, diagnostics, ref resultKind, suppressErrors: attributeType.IsErrorType(), useSiteDiagnostics: ref useSiteDiagnostics);
+                BindAttributeConstructor(node, attributeTypeForBinding, analyzedArguments.ConstructorArguments, overloadResolutionDiagnostics, ref resultKind, suppressErrors: attributeType.IsErrorType(), useSiteDiagnostics: ref useSiteDiagnostics);
```

Comment:
`overloadResolutionDiagnostics` is only used in else part of ternary expression. Consider inlining the declaration there rather than allocating always.